bundle/config/mutator/python/apply_python_output.go

```diff
@@ -0,0 +1,200 @@
+					dyn.NewPatternFromPath(valuePath).Append(dyn.AnyKey()),
+					leftRoot,
+				)
+			} else if len(valuePath) == 3 {
+				// Example: "resources.jobs.job_0"
+
+				return deleted.AddPath(valuePath)
+			} else {
+				// Example: "resources.jobs.job_0.tags"
+
+				return updated.AddPath(valuePath)
+			}
+		},
+		VisitInsert: func(valuePath dyn.Path, right dyn.Value) (dyn.Value, error) {
+			if !valuePath.HasPrefix(resourcesPath) {
+				return dyn.InvalidValue, fmt.Errorf("unexpected change at %q (insert)", valuePath.String())
+			}
+
+			// use rightRoot below because it contains result
+
+			if len(valuePath) == 1 {
+				// Example:
+				//
+				// valuePath: "resources"
+				// leftRoot:  {"bundle": ...,                    }
+				// rightRoot: {"bundle": ..., "resources": {...} }
+
+				return right, added.AddPattern(
+					dyn.NewPatternFromPath(valuePath).Append(dyn.AnyKey(), dyn.AnyKey()),
+					rightRoot,
+				)
+			} else if len(valuePath) == 2 {
+				// Example:
+				//
+				// valuePath: "resources.jobs"
+				// leftRoot:  {"resources": {               }}
+				// rightRoot: {"resources": { "jobs": {...} }}
+
+				return right, added.AddPattern(
+					dyn.NewPatternFromPath(valuePath).Append(dyn.AnyKey()),
+					rightRoot,
+				)
+			} else if len(valuePath) == 3 {
+				// Example:
+				//
+				// valuePath: "resources.jobs"
+				// leftRoot:  {"resources": { "jobs": {               }}}
+				// rightRoot: {"resources": { "jobs": {"job_0": {...} }}}
+
+				return right, added.AddPath(valuePath)
+			} else {
+				// Example: "resources.jobs.job_0.email_notifications"
+
+				return right, updated.AddPath(valuePath)
+			}
+		},
+		VisitUpdate: func(valuePath dyn.Path, _, right dyn.Value) (dyn.Value, error) {
+			if !valuePath.HasPrefix(resourcesPath) {
+				return dyn.InvalidValue, fmt.Errorf("unexpected change at %q (update)", valuePath.String())
+			}
+
+			// use rightRoot below because it contains result
+
+			if len(valuePath) == 1 {
+				// Example:
+				//
+				// valuePath: "resources"
+				// leftRoot:  {"bundle": ..., "resources": null  }
+				// rightRoot: {"bundle": ..., "resources": {...} }
+
+				return right, added.AddPattern(
+					dyn.NewPatternFromPath(valuePath).Append(dyn.AnyKey(), dyn.AnyKey()),
+					rightRoot,
+				)
+			} else if len(valuePath) == 2 {
+				// Example:
+				//
+				// valuePath: "resources.jobs"
+				// leftRoot:  {"resources": { "jobs": null  }}
+				// rightRoot: {"resources": { "jobs": {...} }}
+
+				return right, added.AddPattern(
+					dyn.NewPatternFromPath(valuePath).Append(dyn.AnyKey()),
+					rightRoot,
+				)
+			} else if len(valuePath) == 3 {
+				// Example:
+				//
+				// valuePath: "resources.jobs.job_0"
+				// leftRoot:  {"resources": { "jobs": {"job_0": null  }}}
+				// rightRoot: {"resources": { "jobs": {"job_0": {...} }}}
+
+				return right, added.AddPath(valuePath)
+			} else {
+				// Example: "resources.jobs.job_0.name"
+
+				return right, updated.AddPath(valuePath)
```

Comment:
The AddPath function (that automatically extract resource.A.B.bla.bla from arbitrary paths) does not feel necessary. Why not just add `ResourceKey{valuePath[1], valuePath[2]}` directly?